salt/utils/network.py

```diff
@@ -208,7 +208,7 @@ def get_fqhostname():
         )
         for info in addrinfo:
             # info struct [family, socktype, proto, canonname, sockaddr]
-            if len(info) >= 4:
+            if len(info) >= 4 and info[3]:
```

Comment:
I'm not quite sure what this check is for. Can you explain this a little more?

Reply:
`canonname` can be an empty string. We don't want to populate the `l` list with empty names. @twangboy , maybe add a comment about this?

Reply:
It's making sure there's a value in info[3]. It's sometimes `''` which causes the function to return None on Windows.

Reply:
OK, yes - let's get a comment in here so it doesn't get removed in the future.